models/pull.go

```diff
@@ -1402,3 +1402,36 @@ func TestPullRequests() {
 func InitTestPullRequests() {
 	go TestPullRequests()
 }
+
+// CloseActivePullRequests closes the issue of all the active pull requests associated with a branch
+func CloseActivePullRequests(doer *User, repo *Repository, branchName string) {
+	baseprs, err := GetUnmergedPullRequestsByBaseInfo(repo.ID, branchName)
+	if err != nil {
+		log.Error(4, "Find pull requests [base_repo_id: %d, base_branch: %s]: %v", repo.ID, branchName, err)
+		return
+	}
+	headprs, err := GetUnmergedPullRequestsByHeadInfo(repo.ID, branchName)
+	if err != nil {
+		log.Error(4, "Find pull requests [head_repo_id: %d, head_branch: %s]: %v", repo.ID, branchName, err)
+		return
+	}
+	prs := append(baseprs, headprs...)
+
+	for _, pr := range prs {
+		err = pr.LoadIssue()
+		if err != nil {
+			log.Error(4, "LoadIssue: %v", err)
+			return
+		}
+		err = pr.Issue.ChangeStatus(doer, true)
+		if err != nil {
+			if IsErrDependenciesLeft(err) {
```

Comment:
Hmm. I wonder if we could we avoid this by sorting the PRs by whether they depend on each other? Or would that be too complex/unnecessary?

Reply:
@zeripath I am not sure if I completely understand the comment here.

Reply:
Example: PR A depends on PR B

Both are targeting the same branch.

In this case PR B should be closed before PR A.

Reply:
Thanks a lot for explanation @jonasfranz!
@zeripath It makes sense, I have made the changes. :)